examples/stm32f7disco-qspi-flash/main.rs

```diff
@@ -0,0 +1,135 @@
+//! Example code showing how to use the MT25QL128ABA on the STM32F746G Discovery Board.
+//! The intended behavior of the example is to write a known pattern to flash memory and
+//! read it back using both polling and DMA indirect modes of the QSPI HAL driver.
+//! The example will panic on failure and print messages over the debugger on success.
+//! See `mt25q.rs` for more details on the QSPI HAL driver.
+
+#![no_main]
+#![no_std]
+
+extern crate panic_semihosting;
```

Comment:
These days it's usually better to use RTT together with cargo-embed instead, since semihosting is so slow. No big deal if it works, I'm just saying there are better options.